Update leader election pattern (#120)
mspnp/cloud-design-patterns: leader-election/Readme.md

    ```output
    [14:22:30] This process (51635) is currently the leader. Press any key to exit.
    ```

    If a worker process is not the leader, you will see:

    ```output
    [14:23:21] This process (51686) could not acquire lease. Retrying in 20 seconds. Press any key to exit.
    ```

1. Observe leader election recovery

    You can terminate the current leader and watch one of the other worker processes acquire the lease and become the new leader:

    ```output
    [14:24:21] This process (51686) could not acquire lease. Retrying in 20 seconds. Press any key to exit.
    [14:24:41] This process (51686) could not acquire lease. Retrying in 20 seconds. Press any key to exit.
    [14:25:01] This process (51686) is currently the leader. Press any key to exit.
    ```

1. Run from multiple machines

    If you are using an Azure Storage account, you can run the worker process from multiple machines. Try running it on a second machine, and then temporarily disable or unplug the network on the machine that hosts the leader process. You will see that as the lease expires, one of the still-connected worker processes will claim the lease and assume the role of leader.

> 1. Observer detailed tracing

Reviewer: ```suggestion
1. Observe detailed tracing
```

```shell
git clone https://github.com/mspnp/cloud-design-patterns.git
cd cloud-design-patterns
cd leader-election
```

### Build the solution

Using the existing terminal window, build the solution.

```shell
dotnet build
```

### Running with Azurite storage emulator

The default `app.config` file is set up to use a local Azure Storage emulator. Open a new terminal window, navigate to an empty working directory for the Azurite data files, and start the emulator with the command `azurite`, or `npx azurite` if you installed via `npm`.

### Running with Azure Storage account

Using the Azure portal, create a new Azure Storage account, or us an existing account.

The sample defaults to use a container named `leases`. If you are utilizing an existing storage account, be certain that this container is not already in use.

Find the connection string in your Storage Account's `Access keys` section in the portal. Copy the Connection string value from either key1 or key2 and replace the `StorageConnectionString` value field in `LeaderElectionConsoleWorker/app.config` with the copied value.

### Running the worker process

> Navigate to the `leader-election/LeaderElectionConsoleApp` directory, and start the application

Reviewer: ```suggestion
Navigate to the `leader-election/LeaderElectionConsoleWorker` directory, and start the application.
```

```shell
cd LeaderElectionConsoleWorker
dotnet run
```

> You can start multiple instances of the worker process by opening additional terminal windows or tabs, and running the same command. When a worker process is the leader, you will see periodic output like:

Reviewer: ```suggestion
Ideally, you should start multiple instances of the worker process to see the coordination. To do this, open additional terminal windows or tabs, and run the same command. When a worker process is the leader, you will see periodic output like:
```

```output
[14:22:30] This process (51635) is currently the leader. Press any key to exit.
```

If a worker process is not the leader, you will see:

```output
[14:23:21] This process (51686) could not acquire lease. Retrying in 20 seconds. Press any key to exit.
```

> You can terminate the current leader, and watch the other processes to see one of them acquire the lease and become the new leader:

Reviewer: ```suggestion
You can terminate the current leader and watch one of the other worker processes acquire the lease and become the new leader:
```

```output
[14:24:21] This process (51686) could not acquire lease. Retrying in 20 seconds. Press any key to exit.
[14:24:41] This process (51686) could not acquire lease. Retrying in 20 seconds. Press any key to exit.
[14:25:01] This process (51686) is currently the leader. Press any key to exit.

> ```

Reviewer: Can you tweak this to a bit more closely follow the readme pattern presented on https://github.com/mspnp/cloud-design-patterns/tree/main/pipes-and-filters

Intro (including summary of what it does)
Deployment guide
Prerequisites (bullets)
Steps (numbered)
Try it out (numbered)
Clean up resources (if they selected to use the Azure storage account path)
Related documentation
Contributions

Reviewer: Looks great!
